environments/pettingzoo_atari_adapter.py

```diff
@@ -0,0 +1,323 @@
+        action_space = self.env_specs.get_action_space()
+
+        # Reset environment
+        pz_env.reset(seed=session_cfg.seed)
+        pz_agent_iterator = iter(pz_env.agent_iter())
+        pz_observation, _, _, _, _ = pz_env.last()
+
+        if (len(actor_names) != len(pz_env.agents) + 1) and len(actor_names) > 1:
+            raise ValueError(f"Number of actors does not match environments requirement ({len(pz_env.agents)} actors)")
+        pz_player_names = {agent_name: count for (count, agent_name) in enumerate(pz_env.agents)}
+        pz_player_name = next(pz_agent_iterator)
+        rl_actor_idx = pz_player_names[pz_player_name]
+        actor_name = actor_names[rl_actor_idx]
+
+        # Render the pixel for UI
+        rendered_frame = None
+        if session_cfg.render:
+            if "rgb_array" not in pz_env.metadata["render_modes"]:
+                log.warning(f"Petting Zoo environment [{self.env_class_name}] doesn't support rendering to pixels")
+                return
+            rendered_frame = pz_env.render()
+        observation = observation_space.create(
+            value=pz_observation,
+            rendered_frame=rendered_frame,
+            current_player=actor_name,
+            game_player_name=actor_name,
+            feedback_required=True,
+            action_value=0,
+            step=0,
+        )
+
+        environment_session.start([("*", observation_space.serialize(observation))])
+
+        is_rl_agent = True
+        eval_done = False
+        rewarded_actor_name = actor_name
+        step = 0
+        async for event in environment_session.all_events():
+            if event.actions:
+                # Action
+                action_value = event.actions[rl_actor_idx].action
+                if is_rl_agent:
+                    # Observation
+                    gym_action = action_space.deserialize(action_value)
+                    pz_env.step(gym_action.value)
+                    pz_observation, pz_reward, done, _, _ = pz_env.last()
+
+                    # Actor names for evaluator
+                    rewarded_actor_name = actor_name
+                    rl_actor_idx = -1
+                    actor_name = actor_names[rl_actor_idx]
+
+                    # Pixel frame display on UI
+                    rendered_frame = None
+                    if session_cfg.render:
+                        rendered_frame = pz_env.render()
+                    is_rl_agent = False
+                    eval_done = False
+                    step += 1
+
+                else:
+                    pz_reward = deserialize_ndarray(action_value.value)[0]
+                    print(f"feedback: {pz_reward}")
+                    pz_player_name = next(pz_agent_iterator)
+                    rl_actor_idx = pz_player_names[pz_player_name]
+                    actor_name = actor_names[rl_actor_idx]
+                    eval_done = True
+                    is_rl_agent = True
```

Comment:
Useless, the reward can be sent directly by the human actor

Reply:
@cloderic How could I send the reward from the [front end](https://github.com/cogment/cogment-verse/blob/dev_ppo_pz/cogment_verse/web/web_app/src/controls/AtariPongPzFeedback.jsx#L104)? The only thing I found in the doc is [this ](https://cogment.ai/docs/reference/javascript#class-actorsession-extends-session)